version: "3.4"

services:

  diag01:
    image: lsstts/develop-env:salobj4_b38
    container_name: diag01
    environment:
      - LSST_DDS_DOMAIN=$LSST_DDS_DOMAIN
    volumes:
      - $HOME/develop_diag01:/home/saluser/develop
      - ./diag01_notebook_config.py:/home/saluser/.jupyter/jupyter_notebook_config.py
      - ./diag01_setup.sh:/home/saluser/.setup.sh
    stdin_open: true
    tty: true
    networks:
      control-network:
        ipv4_address: "192.168.1.201"

  diag02:
    image: lsstts/develop-env:salobj4_b38
    container_name: diag02
    environment:
      - LSST_DDS_DOMAIN=$LSST_DDS_DOMAIN
    volumes:
      - $HOME/develop_diag02:/home/saluser/develop
      - ./diag02_notebook_config.py:/home/saluser/.jupyter/jupyter_notebook_config.py
      - ./diag02_setup.sh:/home/saluser/.setup.sh
    stdin_open: true
    tty: true
    networks:
      control-network:
        ipv4_address: "192.168.1.202"

  diag03:
    image: lsstts/develop-env:salobj4_b38
    container_name: diag03
    environment:
      - LSST_DDS_DOMAIN=$LSST_DDS_DOMAIN
    volumes:
      - $HOME/develop_diag03:/home/saluser/develop
      - ./diag03_notebook_config.py:/home/saluser/.jupyter/jupyter_notebook_config.py
      - ./diag03_setup.sh:/home/saluser/.setup.sh
    stdin_open: true
    tty: true
    networks:
      control-network:
        ipv4_address: "192.168.1.203"

  diag04:
    image: lsstts/develop-env:salobj4_b38
    container_name: diag04
    environment:
      - LSST_DDS_DOMAIN=$LSST_DDS_DOMAIN
    volumes:
      - $HOME/develop_diag04:/home/saluser/develop
      - ./diag04_notebook_config.py:/home/saluser/.jupyter/jupyter_notebook_config.py
      - ./diag04_setup.sh:/home/saluser/.setup.sh
    stdin_open: true
    tty: true
    networks:
      control-network:
        ipv4_address: "192.168.1.204"

networks:
  default:
    external:
      name: control-network
  control-network:
    external: true
